thunder/benchmarks/benchmark_scripts.py

```diff
@@ -0,0 +1,117 @@
+import json
+import os
+import subprocess
+
+from typing import Any
+from datetime import datetime
+from functools import partial
+
+from absl import flags
+from absl import logging
+from absl.testing import absltest, parameterized
+
+flags.DEFINE_string("benchmark_output_dir", default="./benchmark_results", help="Output directory for benchmark JSONs")
+
+
+class Target(absltest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._failed_tests = []
+
+        for fn_name in dir(self):
+            if fn_name.startswith("test"):
+                fn = getattr(self, fn_name)
+                wrapped_fn = partial(self._test_wrapper, fn=fn)
+                setattr(self, fn_name, wrapped_fn)
+
+    # Add timestamp and defer exceptions so we can save the test status.
+    def _test_wrapper(self, *args, fn=None, **kwargs):
+        self._start_time = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+        try:
+            return fn(*args, **kwargs)
+        except Exception as ex:
+            self._failed_tests.append(ex)
+
+    def _get_git_revision(self) -> str:
+        cmd = ["git", "rev-parse", "--short", "HEAD"]
+        return subprocess.check_output(cmd).decode("ascii").strip()
+
+    def _report_benchmark_results(self):
+        out_dir = flags.FLAGS.benchmark_output_dir
+        filename = os.path.join(out_dir, f"{self._name}_{self._start_time}.json")
+        git_revision = self._get_git_revision()
+
+        with open(filename, mode="w") as json_report:
```

Comment:
What does an example output look like?

Reply:
Below an example output, I'll add it in the description of the PR too:

``` json
{
    "test_class": "MistralBenchmark",
    "timestamp": "2024-04-19_10-30-23",
    "head": "0534668a",
    "succeeded": true,
    "iter_time": 93333096.0,
    "max_memory_allocated": 5.098717696,
    "input_len": 4096,
    "batch_size": 1,
    "max_iters": 100,
    "warmup_iters": 20,
    "compile": "eager"
}
```